### DISCUSSION TITLE
Insight into Command "Flags" field

### DISCUSSION THREAD

**Discussion by AsherMaximum** [Sandertv/gophertunnel#116] [Ideas]
I've gained some insight into the Flags field:
```
	// Flags is a combination of flags not currently known. Leaving the Flags field empty appears to work.
	Flags uint16
```

Setting Flags to `0x80` allows the command to be used when cheats aren't enabled

**Comment by Sandertv** [comment]
Hmmm that's... rather odd. We're using this in dragonfly and without cheats enabled, those commands are accessible too. 🤔

**Reply by AsherMaximum** [reply]
What's also interesting is that if you create a command with an enum type, in Creative those show up as options. In Survival, they don't, but tab completion still works.

My command:

```
case *packet.AvailableCommands:
	fmt.Printf("AvailableCommands Event:\n\n")
	//fmt.Printf("Commands: %#v\n\n", p.Commands)
	//fmt.Printf("Constraints: %v\n\n", p.Constraints)
	

	//add "go" command
	var goOverloads []protocol.CommandOverload
	var goParameters []protocol.CommandParameter
	var goServers []string
	var goServer string

	for name, _ := range config.Server {
		goServers = append(goServers, strings.ToLower(name))
		if goServer == ""{
			goServer = strings.ToLower(name)
		}else{
			goServer = goServer+" | "+strings.ToLower(name)
		}
	}

	goParameters = append(goParameters, protocol.CommandParameter{
		Name: "server",
		Type: 0x300075,
		Optional: false,
		Enum: protocol.CommandEnum{
			Type: goServer,
			Options: goServers,
			Dynamic: false,
		},
	})

	goOverloads = append(goOverloads, protocol.CommandOverload{
		Parameters: goParameters,
	})

	goCommand := protocol.Command{
		Name: "go",
		Description: "Transfer to another server",
		Flags: 0x80,
		PermissionLevel: 0,
		//Aliases: []string(nil)
		Overloads: goOverloads,
	}

	p.Commands = append(p.Commands, goCommand)
```
Survival:
![image](https://user-images.githubusercontent.com/1038707/149672740-3551b20b-ce9e-4918-ae8f-c0cc0e02e68c.png)

Creative:
![image](https://user-images.githubusercontent.com/1038707/149672720-d43618be-1bdf-4184-a581-c7b8f032ca61.png)

In Survival, if I press `tab` after entering `cr`, it completes with `creative`, even though the options aren't showing like in Creative.

**Reply by Sandertv** [reply]
Something weird's been going on in general with enums in 1.18. I have yet to look into that.
